consumers/best-practices.md

```diff
@@ -41,6 +41,33 @@ Querying the current price will fail if too much time has elapsed since the last
 The SDKs expose this failure condition in an idiomatic way: for example, the Rust SDK may return `None`, and our Solidity SDK may revert the transaction.
 The SDK provides a sane default for the staleness threshold, but users may configure it to suit their use case.
 
+# Latency
+
+Developers integrating Pyth Network price feeds should account for the difference in latency between on-chain oracles and off-chain centralized exchanges.
+Although Pyth Network is designed with low latency in mind, no on-chain oracle can match the latency of an off-chain source due to the added overhead for consensus and security.
+The threat model for integrating protocols should assume that adversaries see price changes a short time before the protocol does.
+In this threat model, protocol designers should avoid situations where a Pyth price update must race against an adversary's transaction.
+The adversary has a head start in these races.
+Furthermore, sophisticated adversaries can highly optimize their network latencies, or pay miners for priority blockspace, such that they are virtually guaranteed to win these races.
+
+This situation is analogous to the one faced by market makers in the traditional finance.
+These market makers place resting orders on exchanges with the hope of earning the bid/ask spread.
+When the “true price” moves, these market makers get picked off by adverse “smart flow” that is faster than they are.
+The smart flow is balanced by two-way flow, that is, people wanting to trade for other reasons besides a price change.
+
+This analogy suggests two simple solutions to races:
```

Comment:
(Not for this PR) but I think diagrams illustrating the problem and the two proposed solutions would be helpful here. They will probably make it more likely that integrators actually read this.